model/map/src/main/java/org/keycloak/models/map/user/MapUserProviderFactory.java

```diff
@@ -43,4 +55,18 @@ public String getHelpText() {
         return "User provider";
     }
 
+    @Override
+    public void invalidate(KeycloakSession session, InvalidableObjectType type, Object... params) {
+        if (type == REALM_BEFORE_REMOVE) {
+            session.users().preRemove((RealmModel) params[0]);
```

Comment:
```suggestion
            create(session).preRemove((RealmModel) params[0]);
```

Reply:
Wouldn't this change effectively bypass current legacy infinispan cache? If so do we want to disable it for map-storage tests?

Reply:
Yes, it should be either disabled for the tests or react to the same invalidation event. The argument is that with `session.users()`, the method of system user store is called as opposed to `this` user store which might be realm-specific.

Reply:
Just one note, this will not bypass only the infinispan cache, but also the JPA legacy user store if used together with map realm store.

We should probably make it clear whether we want to allow combining the old store with the new one (meaning using the legacy store for some entities and map for the rest)

Reply:
We cannot allow that. It is either legacy or map.

Reply:
Ok. Then I agree with this suggestion

Reply:
https://github.com/keycloak/keycloak/pull/11339